src/services/data_model_engine.js

```diff
@@ -204,8 +204,17 @@ export default class DataModelEngine {
     if (!bundle) {
       throw new Error('Could not fetch the bundle from the shelterer');
     }
-    await this.entityBuilder.validateBundle(bundle);
+    this.entityBuilder.validateBundle(bundle);
+    await this.verifyBundle(bundle);
     await this.entityRepository.storeBundle(bundle);
     return bundle;
   }
+
+  async verifyBundle(bundle) {
+    const bundleSizeLimit = await this.contractManager.configWrapper.bundleSizeLimit();
+    if (bundle.content.entries.length > bundleSizeLimit) {
+      throw new Error('Bundle size surpasses the limit');
+    }
+    // TODO another on-chain verifications
```

Comment:
IMHO we don't need todo here as we haven't decided what we want to do :)

Reply:
We have more or less decided on validating metadata, but probably ur right.